workflow-templates/im-deploy-on-prem-website.yml

```
@@ -0,0 +1,462 @@
+          dotnet publish -c Release -o ./published_app --no-restore
+      - name: Zip the published app for faster deployment and uploads
+        working-directory: ${{ env.PROJECT_ROOT }}
+        run: |
+          (cd published_app && zip -r ../${{ env.DEPLOY_ZIP }} .)
+
+      # Uncomment this if the site hasn't been created before
+      # This action does work better if the website cert is already installed
+      # - name: Create Web Site
+      #   uses: im-open/iis-site-create@v1.0.0
+      #   with:
+      #     server: '${{ needs.set-vars.outputs.IIS_SERVER }}'
+      #     website-name: '${{ needs.set-vars.outputs.WEBSITE_NAME }}'
+      #     app-pool-name: '${{ needs.set-vars.outputs.APP_POOL_NAME }}'
+      #     website-host-header: '${{ needs.set-vars.outputs.WEBSITE_HOST_HEADER }}'
+      #     website-path: '${{ needs.set-vars.outputs.WEBSITE_PATH }}'
+      #     website-cert-path: '${{ needs.set-vars.outputs.WEBSITE_CERT_PATH }}'
+      #     website-cert-friendly-name: '${{ needs.set-vars.outputs.WEBSITE_CERT_FRIENDLY_NAME }}'
+      #     website-cert-password: '${{ needs.set-vars.outputs.WEBSITE_CERT_PASSWORD }}'
+      #     service-account-id: '${{ needs.set-vars.outputs.IIS_SERVER_SERVICE_USER }}'
+      #     service-account-password: '${{ needs.set-vars.outputs.IIS_SERVER_SERVICE_PASSWORD }}'
+      #     server-public-key: ${{ needs.set-vars.outputs.IIS_SERVER_CERT_PATH }}
+
+      # TODO:  If config transformation is needed for different environments include environment name
+      # For example to perform transformations on a web.config using the web.production.config file:
+      # dotnet publish --configuration Release /p:EnvironmentName=Production
+      - name: Build and Publish
+        working-directory: ${{ env.PROJECT_ROOT }}
+        run: |
+          dotnet build --configuration Release
+          dotnet publish -c Release -o ./published_app --no-restore
+
+      - name: Zip the published site for faster deployment and uploads
+        working-directory: ${{ env.PROJECT_ROOT }}
+        run: |
+          (cd published_app && zip -r ../${{ env.DEPLOY_ZIP }} .)
+
+      - name: Stop App Pool
+        id: app-pool-stop
+        if: always()
+        uses: im-open/app-pool-action@v1.0.0
+        with:
+          action: 'stop'
+          server: ${{ needs.set-vars.outputs.IIS_SERVER }}
+          app-pool-name: ${{ needs.set-vars.outputs.APP_POOL_NAME }}
+          service-account-id: ${{ needs.set-vars.outputs.IIS_SERVER_SERVICE_USER }}
+          service-account-password: ${{ needs.set-vars.outputs.IIS_SERVER_SERVICE_PASSWORD }}
+          server-public-key: ${{ needs.set-vars.outputs.IIS_SERVER_CERT_PATH }}
+
+      - name: Deploy deployment package
+        if: steps.app-pool-stop.outcome == 'success'
+        uses: 'im-open/deploy-on-prem-web-app@v1.0.0'
+        with:
+          server: ${{ needs.set-vars.outputs.IIS_SERVER }}
+          source-zip-file-path: ../${{ env.DEPLOY_ZIP }}
```

Should this be changed? The double `..` would mean it's going up one from the root of the workspace, wouldn't it?
```suggestion
          source-zip-file-path: ./${{ env.DEPLOY_ZIP }}
```

---

It should be the same as what we specify in the zip step: [`zip -r ../${{env.DEPLOY_ZIP}}`](https://github.com/im-practices/.github/blob/a3f9f6f65f032107b2d9acffa0089775c04dd639/workflow-templates/im-deploy-on-prem-website.yml#L334)

Although we may need to set the working directory.

---

In the zip step it sets the working directory to ./published_app so it's inside that folder.  When it zips it up it is putting it one level up at the root.  

So in this step where we're referring to it, we need to grab the package at the root ./published_app